File: src/elixir/svd/lib/svd/models/cylinder_lube_oil_information.ex
defmodule SVD.Models.CylinderLubeOilInformation do
  @moduledoc false

  @fields [
    :main_engine_cylinder_oil_consumption,
    :main_engine_cylinder_oil_type,
    :main_engine_cylinder_oil_rob,
    :main_engine_system_oil_consumption,
    :main_engine_system_oil_type,
    :main_engine_system_oil_rob,
    :auxiliary_engine_lube_oil_consumption,
    :auxiliary_engine_lube_oil_type,
    :auxiliary_engine_lube_oil_rob,
    :total_lube_oil_bunkered,
    :remaining_on_board,
    :feed_rate,
    :consumption,
    :received_during_bunkering
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end

File: src/go/svd/models/cylinder_lube_oil_information.go
package models

// CylinderLubeOilInformation represents the "Cylinder Lube Oil" section of SVD.
type CylinderLubeOilInformation struct {
	RemainingOnBoard        float64 `json:"remainingOnBoard" xml:"RemainingOnBoard" csv:"RemainingOnBoard"`
	FeedRate                float64 `json:"feedRate" xml:"FeedRate" csv:"FeedRate"`
	Consumption             float64 `json:"consumption" xml:"Consumption" csv:"Consumption"`
	ReceivedDuringBunkering float64 `json:"receivedDuringBunkering" xml:"ReceivedDuringBunkering" csv:"ReceivedDuringBunkering"`
}
